File: Pset 2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    //Prompt user for text
    string text = get_string("Text: ");

    //Declare variables
    int letterscount = 0, wordscount = 1, sentencecount = 0, index;
    double lettersper100, sentenceper100;

    for (int i = 0, n = strlen(text); i < n; i++)
        //For ith character(text[i]) of the string, do the following:
    {
        text[i] = (unsigned char) text[i];

        if (isalpha(text[i]) != 0)
        {
            letterscount++;
        }
        else if (isspace(text[i]) != 0)
        {
            wordscount++;
        }
        else if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            sentencecount++;
        }
    }
    //Find the average number of letters and sentences per 100 words
    lettersper100 = (letterscount * 100) / (double)wordscount;
    sentenceper100 = (sentencecount * 100) / (double)wordscount;

    //Use Coleman-Liau formula to find the grade
    index = round(0.0588 * lettersper100 - 0.296 * sentenceper100 - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: Pset 6/readability.py
from cs50 import get_string

text = get_string("Text: ")

#
letterscount = 0
wordscount = 1
sentencecount = 0
index = 0
lettersper100 = 0
sentenceper100 = 0

for i in range(len(text)):
    if text[i].isalpha():
        letterscount += 1

    elif text[i].isspace():
        wordscount += 1

    elif text[i] == '?' or text[i] == '.' or text[i] == '!':
        sentencecount += 1

# print(letterscount, wordscount, sentencecount)
lettersper100 = (letterscount * 100) / wordscount
sentenceper100 = (sentencecount * 100) / wordscount

# Use Cole-Liau formula to find the grade
index = round(0.0588 * lettersper100 - 0.296 * sentenceper100 - 15.8)

if index < 1:
    print("Before Grade 1")

elif index >= 16:
    print("Grade 16+")

else:
    print(f"Grade {index}")
